src/components/explorer/IndividualExperiments.js

```diff
@@ -52,24 +58,29 @@ const ExperimentResultActions = ({ result }) => {
     }, []);
     const onViewCancel = useCallback(() => setViewModalVisible(false), []);
 
-    const resultViewable = VIEWABLE_FILE_FORMATS.includes(result.file_format)
-        || !!VIEWABLE_FILE_EXTENSIONS.find(ext => result.filename.endsWith(ext));
+    const resultViewable = url && (
+        VIEWABLE_FILE_FORMATS.includes(result.file_format) ||
+        !!VIEWABLE_FILE_EXTENSIONS.find(ext => result.filename.endsWith(ext)));
 
-    return <>
+    return <div style={{ whiteSpace: "nowrap" }}>
+        {url ? <>
+            <Tooltip title="Download">
+                <DownloadButton size="small" uri={url}>{""}</DownloadButton>
+            </Tooltip>
+            {" "}
+        </> : null}
         {resultViewable ? <>
-            <Modal
-                title={<span>View: {result.filename}</span>}
+            <FileModal
                 visible={viewModalVisible}
                 onCancel={onViewCancel}
-                width={1080}
-                style={{marginTop: "-50px"}}
-                footer={null}
-            >
-                {hasTriggeredViewModal && (
-                    <FileDisplay uri={url} fileName={result.filename} />
-                )}
-            </Modal>
-            <Button size="small" icon="eye" onClick={onViewClick}>View</Button>{" "}
+                title={<span>View: {result.filename}</span>}
+                url={url}
+                fileName={result.filename}
+                hasTriggered={hasTriggeredViewModal}
+            />
+            <Tooltip title="View">
```

Comment:
The button is not triggering the file download as expected. Could there be a conflict between the tooltip events and the onClick handler?

Reply:
might be the case, i'll check it out!

Reply:
should be fixed now